Source/WebCore/platform/graphics/SourceBufferPrivate.cpp

```diff
@@ -189,10 +189,18 @@ void SourceBufferPrivate::seekToTime(const MediaTime& time)
     }
 }
 
-void SourceBufferPrivate::clearTrackBuffers()
+void SourceBufferPrivate::clearTrackBuffers(bool reportClient)
 {
     for (auto& trackBuffer : m_trackBufferMap.values())
         trackBuffer->clearSamples();
+
+    if (!reportClient)
+        return;
+
+    updateBufferedFromTrackBuffers(true);
+    updateHighestPresentationTimestamp();
+
+    m_client->sourceBufferPrivateReportExtraMemoryCost(totalTrackBufferSizeInBytes());
```

Comment:
Are we sure m_client is always never null?

Reply:
sounds like it could be if called before setClient is called, there are some missing check too, will do some flyby